src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -1095,4 +1120,11 @@ private static boolean hasActiveChild(Project project) {
         }
         return hasActiveChild;
     }
+
+    private List<ProjectVersion> getProjectVersions(Project project) {
+        final Query<Object[]> query = pm.newQuery(JDOQuery.SQL, "SELECT UUID, VERSION FROM PROJECT WHERE NAME = ?");
+        query.setParameters(project.getName());
+        final var stream = query.executeList().stream();
+        return stream.map(i -> new ProjectVersion(i[0].toString(), i[1].toString())).toList();
```

Comment:
You can make use of projections / [result classes](https://www.datanucleus.org/products/accessplatform_6_0/jdo/query.html#resultclass) here. That way DataNucleus will perform the mapping for you, and similar to your SQL it will also only fetch the required fields:

```suggestion
        final Query<Project> query = pm.newQuery(Project.class);
        query.setFilter("name == :name");
        query.setParameters(project.getName());
        query.setResult("uuid, version");
        return query.executeList(ProjectVersion.class);
```